### DISCUSSION TITLE
Manual mesh bed levelling

### DISCUSSION THREAD

**Discussion by yetistu13** [Jyers/Marlin#116] [Q&A] [answered]
I’m using 5x5 manual mesh. V1.1.0

First print goes down perfect. 

Second print won’t. It’s as if it’s not using the mesh data. 

Is there something I need to do in Gcode to enable the mesh ?

**Comment by t3chguy** [comment, marked as answer]
A `Home` operation discards your bed mesh, you may want to use https://marlinfw.org/docs/gcode/M420.html `M420 S1`

**Reply by yetistu13** [reply]
Ok great so I pop that in after a g28. 

I’ve looked at vids that show adjusted the bed z . How do I do that ? Can’t find an option for that.

**Reply by Schnabulation** [reply]
Also make sure to store the mesh data to EEPROM. Otherwise they will get deleted after a reboot of the printer.

**Reply by Jyers** [reply]
> I’ve looked at vids that show adjusted the bed z . How do I do that ? Can’t find an option for that.

The z offset menu can be found in the prepare menu.

**Reply by xantrk** [reply]
> A `Home` operation discards your bed mesh, you may want to use https://marlinfw.org/docs/gcode/M420.html `M420 S1`

Isn't #define RESTORE_LEVELING_AFTER_G28 flag fixes that? Which is already enabled in precompiled firmware?

May I ask how to disable mesh leveling from the menus? If I saved it to EEPROM, is there a way to prevent it from loading again on next boot? @Jyers

**Reply by Schnabulation** [reply]
No, but you could just reset EEPROM.

**Reply by Jyers** [reply]
You can also disable the mesh from a host using M420 S0

**Reply by vaderag** [reply]
Just wanted to follow up on this thread as I had the same and a little confused at the above.

To walk through what I did:
- using pre-compiled 3x3
- I used the Levelling button on the home screen
- Went along and set all my levels, mix of manual and z-offset steps on the screen
- Saved configuration in settings
- Re-leveled to check
- Printed - had a lot of under extrusion points and clicking so had to adjust z up on the fly

Long story short, I either levelled badly (certainly a possibility as was my first time) or I missed a step

So, reading the above I'm just wondering, do I have to add a command to gcode on the first print I do after levelling to actually save it? Or is what I did right and I just did it badly?

**Reply by Sennevds** [reply]
Got the same problem. After manual mesh my pla won't stick. I've added M420 S1 after G28. When I manually level without creating a mesh it's the all right but if I use manual mesh it's way to high for some reason

**Reply by Schnabulation** [reply]
I don't know how it is for the manual mesh, but with a BL Touch you have to set a negative Z-offset (-3mm in my case), otherwise the nozzle is not close enough. Maybe it's the same with the manually created mesh.
